### PR TITLE
windows: deal with the fact that GetTickCount might lag

### PR DESCRIPTION
**Pull request by piscisaureus**
We use GetQueuedCompletionStatus(Ex) to sleep the thread until the next
timer expires (provided that no other events happen before that).
However after waking up from a sleep the GetTickCount() return value may
not immediately reflect that some time has passed. This happens because
gqcs can sometimes sleep for periods shorter than the GetTickCount clock
resulution. This patch changes time tracking so the amount of time
waited by gqcs is taken into account.

This has the following advantages:
- Excessive loop iterations are avoided.
- Small timeouts are fired more precisely.
- The `loop-stop` test that used to be flaky on Windows now passes
  consistently.

@bnoordhuis or @sblom -> review?

### CODE REVIEW

File: src/win/timer.c [hunk 1]
```diff
@@ -29,19 +29,38 @@
...
-  /* If the timer has wrapped, add 1 to it's high-order dword. */
+  /* GetTickCount() can conceivably wrap around, so when the current tick */
+  /* count is lower than the last tick count, we'll assume it has wrapped. */
   /* uv_poll must make sure that the timer can never overflow more than */
   /* once between two subsequent uv_update_time calls. */
-  if (ticks < time->LowPart) {
-    time->HighPart += 1;
-  }
-  time->LowPart = ticks;
+  time = *(ULARGE_INTEGER*) &loop->time;
```
Comment (bnoordhuis) [inline src/win/timer.c:41]: Doesn't msvc complain loudly about aliasing here?

**Comment by bnoordhuis** [comment]
I don't know, I still think it's a deeply evil hack.

File: src/win/timer.c (hunk 1 above)
Comment (piscisaureus) [inline src/win/timer.c:41]: No, complaining about aliasing is a gcc-ism :)

**Comment by piscisaureus** [comment]
@sblom, can I have your opinion here as well. Terrible hack or acceptable solution?

**Pull request closed without merging**